pkg/component/dependencywatchdog/access.go

```diff
@@ -67,42 +98,150 @@ type dependencyWatchdogAccess struct {
+			Name:      "gardener.cloud:target:" + prefixDependencyWatchdog,
+			Namespace: corev1.NamespaceNodeLease,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     role.Name,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      strings.TrimPrefix(KubeConfigSecretName, gardenerutils.SecretNamePrefixShootAccess),
+			Namespace: metav1.NamespaceSystem,
+		}},
 	}
 
-	return nil
+	resources, err := registry.AddAllAndSerialize(
+		role,
+		roleBinding,
+	)
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForShoot(ctx, d.client, d.namespace, managedResourceTargetName, managedresources.LabelValueGardener, false, resources)
+}
+
+// TODO(aaronfern): remove this when g/g v1.91 is released
+func (d *dependencyWatchdogAccess) DeployMigrate(ctx context.Context) error {
+	// Create shoot access secret
+	caSecret := corev1.Secret{}
+	err := d.client.Get(ctx, types.NamespacedName{Namespace: d.namespace, Name: v1beta1constants.SecretNameCACluster}, &caSecret)
+	if err != nil {
+		return fmt.Errorf("error in fetching secret %s. err: %v", v1beta1constants.SecretNameCACluster, err)
+	}
+
+	var (
+		shootAccessSecret = gardenerutils.NewShootAccessSecret(KubeConfigSecretName, d.namespace).WithNameOverride(KubeConfigSecretName)
+		kubeconfig        = kubernetesutils.NewKubeconfig(
+			d.namespace,
+			clientcmdv1.Cluster{Server: d.values.ServerInCluster, CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle]},
+			clientcmdv1.AuthInfo{Token: ""},
+		)
+	)
+
+	if err := shootAccessSecret.WithKubeconfig(kubeconfig).Reconcile(ctx, d.client); err != nil {
+		return err
+	}
+
+	var registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gardener.cloud:target:" + prefixDependencyWatchdog,
+			Namespace: corev1.NamespaceNodeLease,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"get", "list"},
+			},
+		},
+	}
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gardener.cloud:target:" + prefixDependencyWatchdog,
+			Namespace: corev1.NamespaceNodeLease,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     role.Name,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      strings.TrimPrefix(KubeConfigSecretName, gardenerutils.SecretNamePrefixShootAccess),
+			Namespace: metav1.NamespaceSystem,
+		}},
+	}
+
+	resources, err := registry.AddAllAndSerialize(
+		role,
+		roleBinding,
+	)
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForShoot(ctx, d.client, d.namespace, managedResourceTargetName, managedresources.LabelValueGardener, false, resources)
```

Comment:
Isn't this almost the same code as in `Deploy` (except for fetching the CA secret)? If so, don't duplicate it, please, but rather extract it into a function and make it reusable for both places.